sound/soc/sof/sof-audio.c

```diff
@@ -14,11 +14,29 @@
 
 static void sof_reset_route_setup_status(struct snd_sof_dev *sdev, struct snd_sof_widget *widget)
 {
+	const struct sof_ipc_tplg_ops *tplg_ops = sdev->ipc->ops->tplg;
+	struct snd_sof_widget *route_endpoint_widget;
 	struct snd_sof_route *sroute;
 
 	list_for_each_entry(sroute, &sdev->route_list, list)
-		if (sroute->src_widget == widget || sroute->sink_widget == widget)
+		if (sroute->src_widget == widget || sroute->sink_widget == widget) {
+			if (sroute->src_widget == widget)
+				route_endpoint_widget = sroute->sink_widget;
+
+			if (sroute->sink_widget == widget)
+				route_endpoint_widget = sroute->src_widget;
+
+			/* only free the route if the other endpoint widget's use_count is 0 */
+			if (route_endpoint_widget->use_count) {
```

Comment:
maybe that's obvious to you, but it wouldn't to add a comment on where the route_endpoint_widget use count will be decreased.

Reply:
@plbossart this case isnt that obvious to me. What I observe with IPC4 today is that when we have 2 modules belonging to different pipelines  such as a mixin/mixout, we cannot unbind them is the mixout is being used by another pipeline that is still active. I am still confirming what needs to happen in such cases. But I have changed the logic here and explained it a bit better now.

Reply:
on second thoughts, I think this is a special case for IPC4 and I will handle it in its route_free op.